godel_surface_detection/src/scan/profilimeter_scan.cpp

```diff
@@ -0,0 +1,244 @@
+      
+      dx = -std::sin(bbox.a);
+      dy = std::cos(bbox.a);
+
+      x = bbox.x - dx * (bbox.h - adjusted_width) / 2.0;
+      y = bbox.y - dy * (bbox.h - adjusted_width) / 2.0; 
+
+      w = bbox.w;
+      h = adjusted_width;
+    }
+    
+    std::vector<RotatedRect> slices;
+    slices.reserve(n_slices);
+
+    dx *= adjusted_width;
+    dy *= adjusted_width;
+
+    for (int i = 0; i < n_slices; ++i)
+    {
+      RotatedRect temp;
+      temp.a = bbox.a;
+      temp.x = x;
+      temp.y = y;
+      temp.w = w;
+      temp.h = h;
+      slices.push_back(temp);
+      x += dx;
+      y += dy;
+    }
+
+    return slices;
+  }
+
+  std::vector<PolygonPt> interpolateAlongAxis(const RotatedRect& rect, double ds)
+  {
+    std::vector<PolygonPt> pts;
+    double dx, dy; // 
+    double x, y;
+    int n;
+    if (rect.w > rect.h)
+    {
+      n = rect.w / ds;
+      dx = std::cos(rect.a);
+      dy = std::sin(rect.a);
+      x = rect.x - dx * rect.w/2.0;
+      y = rect.y - dy * rect.w/2.0; 
+    }
+    else
+    {
+      n = rect.h / ds;
+      dx = -std::sin(rect.a);
+      dy = std::cos(rect.a);
+      x = rect.x - dx * rect.h/2.0;
+      y = rect.y - dy * rect.h/2.0;
+    }
+
+    // pts.reserve(n);
+
+    // transform unit vector
+    dx *= ds; 
+    dy *= ds;
+
+    for (int i = 0; i < n; ++i)
+    {
+      PolygonPt pt;
+      pt.x = x;
+      pt.y = y;
+      x += dx;
+      y += dy;
+      pts.push_back(pt);
+    }
+
+    return pts;
+  }
+
+  std::vector<PolygonPt> makeStitch(const PolygonPt& a, const PolygonPt& b, double ds)
+  {
+    double dx = b.x - a.x;
+    double dy = b.y - a.y;
+    double s = std::sqrt(dx*dx + dy*dy);
+    // steps
+    size_t n = s / ds;
+    // unit vector
+    dx /= s;
+    dy /= s;
+
+    std::vector<PolygonPt> result;
+    for (size_t i = 0; i < n; ++i)
+    {
+      PolygonPt pt;
+      pt.x = a.x + i * dx * ds;
+      pt.y = a.y + i * dy * ds;
+      result.push_back(pt);
+    }
+    return result;
+  }
+
+  std::vector<PolygonPt> stitchAndFlatten(const std::vector<std::vector<PolygonPt> >& paths, double ds)
+  {
+    std::vector<PolygonPt> result;
+    bool forward = true;
+    for (std::size_t i = 0; i < paths.size(); ++i)
+    {
+      // copy current path in given direction
+      if (forward)
+      {
+        result.insert(result.end(), paths[i].begin(), paths[i].end());
+      }
+      else
+      {
+        result.insert(result.end(), paths[i].rbegin(), paths[i].rend());
+      }
+
+      forward = !forward;
+
+      // check to see if there is a next strip
+      if (i+1 < paths.size())
+      {
+        // if so, stitch to the next point
+        std::vector<PolygonPt> stitch = makeStitch(*result.rbegin(), (forward ? *(paths[i+1].begin()) : *(paths[i+1].rbegin())), ds);
+        result.insert(result.end(), stitch.begin(), stitch.end());
+      }
+    }
+
+    return result;
+  }
+
+
+} // end anon namespace
+
+void printPt(const PolygonPt& pt)
```

Comment:
Having a function that just prints a warning feels unjustified. Plus it doesn't seem to be invoked elsewhere in the scope of this source file